sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/UcfgInstructionFactory.cs

```diff
@@ -336,10 +361,62 @@ private Expression GetTargetExpression(SyntaxNode syntaxNode, ISymbol nodeSymbol
             return expressionService.CreateClassName(nodeSymbol.ContainingType);
         }
 
-        private IEnumerable<Instruction> ProcessArrayCreation(ArrayCreationExpressionSyntax arrayCreation)
+        private IEnumerable<Instruction> ProcessArrayCreation(ArrayCreationExpressionSyntax arrayCreationExpression)
         {
-            var arrayTypeSymbol = semanticModel.GetTypeInfo(arrayCreation).Type as IArrayTypeSymbol;
-            return BuildNewInstance(arrayCreation, arrayTypeSymbol);
+            var typeInfo = semanticModel.GetTypeInfo(arrayCreationExpression);
+            var arrayTypeSymbol = typeInfo.Type as IArrayTypeSymbol ??
```

Comment:
I don't know if it's possible with an _ArrayCreationExpressionSyntax_, but it is with the syntax _string[] a2 = new[] { data, data }_. I decided to err on the side of caution in case there were any other cases were the type was null.
There's also a case in which we don't necessarily want the conversion (object initializer of where the parameter is a string array), although I think I have commented the location where that can occur.